dask/dataframe/core.py

```diff
@@ -304,31 +348,69 @@ class _Frame(DaskMethodsMixin, OperatorMethodMixin):
 
     Parameters
     ----------
-    dsk: dict
+    dsk: dict, Optional
         The dask graph to compute this DataFrame
-    name: str
+    name: str, Optional
         The key prefix that specifies which keys in the dask comprise this
         particular DataFrame / Series
-    meta: pandas.DataFrame, pandas.Series, or pandas.Index
+    meta: pandas.DataFrame, pandas.Series, or pandas.Index, Optional
         An empty pandas object with names, dtypes, and indices matching the
         expected output.
-    divisions: tuple of index values
+    divisions: tuple of index values, Optional
         Values along which we partition our blocks on the index
+    operation: FrameOperation, Optional
+        Instead of initializing a ``_Frame`` object with ``dsk``, ``name``,
+        ``meta``, and ``divisions``, the state of the collection can be
+        defined by a ``FrameOperation`` object.
     """
 
-    def __init__(self, dsk, name, meta, divisions):
-        if not isinstance(dsk, HighLevelGraph):
-            dsk = HighLevelGraph.from_collections(name, dsk, dependencies=[])
-        self.dask = dsk
-        self._name = name
-        meta = make_meta(meta)
-        if not self._is_partition_type(meta):
+    def __init__(self, dsk=None, name=None, meta=None, divisions=None, operation=None):
+        if operation is None:
+            if any(v is None for v in [dsk, name, meta]):
+                raise ValueError(
+                    "Must specify dsk, name, and meta if " "operation is not specified."
+                )
+            self._operation = CompatFrameOperation(
+                dsk,
+                name,
+                meta=make_meta(meta),
+                divisions=divisions,
+            )
+        else:
+            if not isinstance(operation, FrameOperation):
+                raise ValueError(f"Expected FrameOperation, got {type(operation)}")
+            elif any([dsk, name, meta, divisions]):
+                raise ValueError(
+                    "dsk, name, meta, and divisions can not be passed "
+                    "to _Frame if operation is specified."
+                )
+            self._operation = operation
+
+        if not self._is_partition_type(self._meta):
             raise TypeError(
                 f"Expected meta to specify type {type(self).__name__}, got type "
-                f"{typename(type(meta))}"
+                f"{typename(type(self._meta))}"
             )
-        self._meta = meta
-        self.divisions = tuple(divisions)
+
+    @property
+    def operation(self):
+        return self._operation
+
+    @property
+    def dask(self):
+        return self.operation.dask
+
+    @property
+    def _name(self):
+        return self.operation.name
+
+    @property
+    def _meta(self):
+        return self.operation.meta
+
+    @_meta.setter
+    def _meta(self, value):
+        self._operation = self.operation.replay(meta=value)
```

Comment:
Note that this will not change the name of the new operation (and therefore the collection). However, the `FrameOperation` will be hashed using `name`/`meta`/`divisions`, so it is still important that we "replay" operations when a property is changed.

Note that I originally wanted to require the operations name to uniquely depend on its perperties, but this doesn't work for `CompatFrameOperation` (where we cannot change the name used in the existing HLG).